samples/snippets/csharp/VS_Snippets_CLR/HookUpDelegate/cs/source.cs

```diff
@@ -96,7 +96,7 @@ private void HookUpDelegate()
         //<Snippet9>
         Type returnType = GetDelegateReturnType(tDelegate);
         if (returnType != typeof(void))
-            throw new ApplicationException("Delegate has a return type.");
+            throw new Exception("Delegate has a return type.");
```

Comment:
I think it's also a bad practice to throw the general exception type. Probably `InvalidOperationException` is more suitable?

Reply:
You are right, maybe even an `ArgumentException` since most of the checks are done on the argument, or properties on the argument?

Reply:
I like the `ArgumentException` here.